code/modules/projectiles/projectile/bullets/sniper.dm

```diff
@@ -1,56 +1,99 @@
-// .50 (Sniper)
+// .50 BMG (Sniper)
 
 /obj/projectile/bullet/p50
-	name =".50 bullet"
+	name =".50 BMG bullet"
 	speed = 0.4
 	range = 400 // Enough to travel from one corner of the Z to the opposite corner and then some.
 	damage = 70
 	paralyze = 100
 	dismemberment = 50
+	catastropic_dismemberment = TRUE
 	armour_penetration = 50
-	var/breakthings = TRUE
+	///Determines object damage.
+	var/object_damage = 80
+	///Determines how much additional damage the round does to mechs.
+	var/mecha_damage = 10
 
 /obj/projectile/bullet/p50/on_hit(atom/target, blocked = 0)
```

Comment:
restructure this to be like
dealt damage = object damage
if ismecha target
dealt damage += mecha damge
if dealt damage
take_damage

then you dont need to copypaste the code on the disruptor